ipywidgets/widgets/widget_media.py

```diff
@@ -165,6 +165,49 @@ def __repr__(self):
         return self._get_repr(Image)
 
 
+class MapArea(object):
+    r"""
+    An <area> tag, part of an image map <map>.
+    """
+    def __init__(self, name, shape, coords, href=None):
+        """
+        Parameters
+        ----------
+        name: str
+            Area name. Will be used as 'alt' attribute.
+
+        shape: str
+            Area shape: rect/circle/poly/default.
+
+        coords: tuple
+            Area coordinates: tuple of integers.
+
+        href: str
+            URL to load when area is clicked.
+        """
+        self.name = name
+        self.shape = shape
+        self.coords = coords # a tuple of integers
+        self.href = href
+
+    def html(self):
+        return  '<area alt="{}" shape="{}" coords="{}" href="{}">' . format(
+            self.name,
+            self.shape,
+            ",". join([str(i) for i in self.coords]),
+            self.href
+        )
+
+
+@register
+class MappedImage(Image):
+    _view_name = Unicode('MappedImageView').tag(sync=True)
+    _model_name = Unicode('MappedImageModel').tag(sync=True)
+
+    map_name = Unicode("Map", help="The map name").tag(sync=True)
```

Comment:
I wonder if the `map_name` should be exposed to the user, as far as I understand this is an implementation detail and does not bring anything to the Python API.

Maybe we should set the map name to a random uuid? And not expose it to Python?

Reply:
Thanks @martinRenou 
Yes, traits.
Yes, why not not expose map_name.
At the moment, I experience some difficulties in displaying the map areas. I could possible create a new HTMLElement widget, that would be like HTML, without its surrounding `<div>`.  Don't you think such an element would bring some flexibility?

Reply:
I am not sure I understand what you are saying.

But looking at the code I understand the `ImageView` implementation should be changed in order for this to work, `this.el` should be an `HTMLDivElement` instead of an `HTMLImageElement`. And it would contain the `HTMLImageElement` and the `HTMLMapElement`

Reply:
I hesitate.
I guess having an `<img>` without `<div>` is there for a reason, in Image widget.
But I would like to use this `HTMLElement` to create and display the map areas, and then the map itself.
Maybe such a strategy is too complicated. I want to test these things anyway, but if you tell me such an `HTMLElement` (with arbitraty tag name, not necessarily 'div') would be useful / or not useful, that will help guiding my efforts anyway.

Reply:
I am not talking about a new `HTMLElement` widget. I am talking about the html tags.

I was suggesting doing:
```
<div>
    <img src="...">
    <map>
        <area ...>
        <area ...>
    </map>
</div>
```

instead of the current:
```
<img src="...">
```

In the `Image` widget